### PR TITLE
307 watcher incorrect handling of failed job

### PR DESCRIPTION
**Pull request by mbthornton-lbl**
This PR provides provides updates and tests to ensure that a job from the state file with < MAX_FAILURES will be picked up and re-submitted

Changes:
- Update `get_finished_jobs` to check and update job status for unfinished jobs
- Unit test to verify correct behavior
- Unit test to verify that 1x failed job gets re-submitted

Also update .coveragerc to exclude obsolete `re_iding` package, and update badges and .xml.
These are not getting automatically updated by the CI build

  
Log from running on Permutter in dev showing pickup and re-submit of 1x failed job:
```
2024-11-26 13:04:39,444 INFO: Initializing Watcher: config file: /global/homes/n/nmdcda/nmdc_automation/dev/site_configuration_nersc.toml
2024-11-26 13:04:39,445 INFO: Using state file from config: /global/cfs/cdirs/m3408/var/dev/agent.state
2024-11-26 13:04:39,445 INFO: New Job from State: nmdc:wfmag-11-g7msr323.1, nmdc:66cf64b6-7462-11ef-8b84-deaa01ab0f49
2024-11-26 13:04:39,445 INFO: Last Status: Succeeded
2024-11-26 13:04:39,445 INFO: New Job from State: nmdc:wfmag-12-h52r0792.1, nmdc:c2b7c884-ab78-11ef-8298-3e652b5abb3d
2024-11-26 13:04:39,445 INFO: Last Status: Failed
2024-11-26 13:04:39,445 INFO: Adding 2 new jobs from state file.
2024-11-26 13:04:40,575 INFO: Entering polling loop
2024-11-26 13:04:41,418 INFO: Found 0 unclaimed jobs.
2024-11-26 13:04:41,420 INFO: Checking for finished jobs.
2024-11-26 13:04:41,420 INFO: Found 1 failed jobs.
2024-11-26 13:04:41,420 INFO: Processing failed job: nmdc:sys0a5361811, nmdc:wfmag-12-h52r0792.1
2024-11-26 13:04:41,422 ERROR: Job nmdc:sys0a5361811 failed 2 times. Retrying.
2024-11-26 13:04:43,137 INFO: Submitted job 2c0212ad-83d3-4cc5-85cc-8e6434cb21f8
2024-11-26 13:04:43,137 INFO: Job 2c0212ad-83d3-4cc5-85cc-8e6434cb21f8 submitted
```

### CODE REVIEW

File: nmdc_automation/workflow_automation/watch_nmdc.py [hunk 1]
```diff
@@ -252,8 +278,9 @@ def process_failed_job(self, job: WorkflowJob) -> None:
         job.workflow.state["failed_count"] = job.workflow.state.get("failed_count", 0) + 1
         job.workflow.state["last_status"] = job.job_status
         self.save_checkpoint()
-        logger.error(f"Job {job.opid} failed {job.workflow.state['failed_count']} times. Retrying.")
-        job.job.submit_job()
+        logger.info(f"Job {job.opid} failed {job.workflow.state['failed_count']} times. Retrying.")
```
Comment (shreddd) [inline nmdc_automation/workflow_automation/watch_nmdc.py:281]: should this be logger.warn?

File: nmdc_automation/workflow_automation/watch_nmdc.py [hunk 2]
```diff
@@ -292,7 +319,7 @@ def update_operation(self, opid, done, meta):
 class Watcher:
     """ Watcher class for monitoring and managing jobs """
     def __init__(self, site_configuration_file: Union[str, Path],  state_file: Union[str, Path] = None):
-        self._POLL = 20
+        self._POLL = 60
```
Comment (shreddd) [inline nmdc_automation/workflow_automation/watch_nmdc.py:322]: Could you clarify what this interval is (seconds I presume) in the comments?

**Review by shreddd — COMMENTED** [review]
Made some minor comments. Probably simpler to do a walkthrough or summary of the intent of this code update to clarify what I should be looking at for review.

File: nmdc_automation/workflow_automation/watch_nmdc.py (hunk 1 above)
Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/watch_nmdc.py]: changed to warning

File: nmdc_automation/workflow_automation/watch_nmdc.py (hunk 2 above)
Comment (mbthornton-lbl) [reply nmdc_automation/workflow_automation/watch_nmdc.py]: Renamed prop

**Review by aclum — APPROVED** [review]

**Pull request merged**